File: server/Domain/Idea.cs
using System;

namespace Domain
{
    public class Idea
    {
        public Guid Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public string Category { get; set; }

        public DateTime Created { get; set; }

        public DateTime Updated { get; set; }
    }
}

File: client/ui-app/src/app/models/idea.ts
export interface IIdea {
  id: string;
  title: string;
  description: string;
  category: string;
  created: Date;
  updated: Date;
}

export interface IIdeaFormValues extends Partial<IIdea> {
  time?: Date;
}

export class IdeaFormValues implements IIdeaFormValues {
  id?: string = undefined;
  title: string = "";
  category: string = "";
  description: string = "";
  created?: Date = undefined;
  time?: Date = undefined;
  updated?: Date = undefined;

  constructor(init?: IIdeaFormValues) {
    if (init && init.created) {
      init.time = init.created;
    }
    Object.assign(this, init);
  }
}
